### PR TITLE
Don't show warning screen when registering on dev server

### PR DESCRIPTION
**Pull request by Joehuu**

### CODE REVIEW

**Comment by Joehuu** [comment]
Didn't see the `load()` method have `OsuGame`. Will put out.

**Comment by peppy** [comment]
The problem with making changes like this is it means this flow will never get tested by devs any more.

I'm not sure if we want to do this or not. That said, I'm pretty neutral on the issue so will wait for further opinions.

**Comment by smoogipoo** [comment]
I'm pretty neutral on this too, but I don't think the "never get tested by devs any more" argument is super valid. I don't need to create any more accounts so can't confirm that this screen will never break, but also this doesn't apply in tests (due to the `OsuGame` dependency that doesn't exist in tests) and I think this is better tested there.

**Comment by frenzibyte** [comment]
This might be a duplicate of what's said above, but, writing a test scene for this that goes through the registering flow and ensuring the right screen is pushed would do better I think.

I'm not really fussed with hiding the warning screen, but I think this is the right job for test scenes.

File: osu.Game/Overlays/AccountCreation/ScreenWarning.cs [hunk 1]
```diff
@@ -26,11 +26,14 @@ public class ScreenWarning : AccountCreationScreen
...
 
+        [Resolved(canBeNull: true)]
+        private OsuGame game { get; set; }
+
         private const string help_centre_url = "/help/wiki/Help_Centre#login";
 
         public override void OnEntering(IScreen last)
         {
-            if (string.IsNullOrEmpty(api?.ProvidedUsername))
+            if (string.IsNullOrEmpty(api?.ProvidedUsername) || game.UseDevelopmentServer)
```
Comment (bdach) [inline osu.Game/Overlays/AccountCreation/ScreenWarning.cs:36]: If nullability of `game` is to be allowed (which it wasn't before), then this should be

```suggestion
            if (string.IsNullOrEmpty(api?.ProvidedUsername) || game?.UseDevelopmentServer)
```

lest it crash on a `null`.

**Review by bdach — CHANGES_REQUESTED** [review]
I *also* don't really have a strong opinion on this - I suppose I am weakly in favour of doing this, to avoid answering the same questions from new contributors about this dialog in context of the dev environment. But I *do* think this ought to have comprehensive tests - as to why, see suggestion below, which was not caught.

**Review by peppy — APPROVED** [review]

**Pull request merged**